tests/integration/managed_array_tests.cpp

```diff
@@ -593,7 +591,48 @@ TEST(ManagedArray, ReallocateCPU)
   assert_empty_map(true);
 }
 
+TEST(ManagedArray, ReallocateCopyCPU)
```

Comment:
Can you provide more context about what this test case is for? I think in general after a reallocate of the original array, any copies made prior to the reallocate are invalidated. The copies might get back in sync after a move is triggered (since they share the same pointer record) in the implementation in ManagedArray.inl, but I don't think this is the case in the thin version where a pointer record is not used.

Reply:
There is no functionality to "invalidate" a copy of a ManagedArray nor query if it has been invalidated. I strongly believe any copy of a ManagedArray should maintain it's relationship to the underlying allocations it is pointing to, this upholds proper reference semantics copying and is why pointer records exist.

This tests the behavior changed in the `size` member function of a MA seen above. Perhaps this points to a disconnect between the expected behavior of a MA and a gap in the interface. Chai needs to either be explicit about invalidating copies upon reallocation or maintain correct relationships between.